### PR TITLE
adds "run-script post-deploy-cmd" to composer.sh and adds tests

### PR DESCRIPTION
**Pull request by bshaffer**

### CODE REVIEW

**Comment by bshaffer** [comment]
see #135

**Comment by tmatsuo** [comment]
Some scripts which needs modification to the base image will fail.

I think we need to run `composer install` 2 times as described at: https://github.com/GoogleCloudPlatform/php-docker/issues/135#issuecomment-237001254

**Comment by bshaffer** [comment]
@tmatsuo @cedricziel I added support for running custom scripts in `entrypoint.sh` for `post-deploy-cmd`. PTAL.

File: php-nginx/entrypoint.sh [hunk 1]
```diff
@@ -114,4 +114,16 @@ sed -i "s|%%DOC_ROOT%%|${DOCUMENT_ROOT}|g" "${PHP_DIR}/lib/php.ini"
...
 ${PHP_DIR}/bin/php /whitelist_functions.php
 
+# run the composer scripts for post-deploy
+COMPOSER_CMD="${PHP_DIR}/bin/php \
+    -d suhosin.executor.include.whitelist=phar \
+    -d suhosin.executor.func.blacklist=none \
+    -d disable_functions= \
+    -d memory_limit=-1 \
+    -d max_input_time=-1 \
+    /usr/local/bin/composer"
```
Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: `--no-ansi` ?

File: testapps/php56_custom/TestScript.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+namespace PhpDocker\TestApps\Php56Custom;
+
+use Symfony\Component\ClassLoader\ClassCollectionLoader;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Process\PhpExecutableFinder;
```
Comment (tmatsuo) [inline testapps/php56_custom/TestScript.php]: Are they used?

**Comment by tmatsuo** [comment]
LGTM

But let's wait for @cedricziel for his opinion.

This change won't affect your stuff because we're using custom script name.

Then we can say "To customize further, add a composer script named 'post-deploy-cmd'" in our guides.

How does it sound?

**Comment by cedricziel** [comment]
Sounds good! :+1:

**Comment by tmatsuo** [comment]
@bshaffer 
I think it's ready. Can I merge this? I will create another PR for running composer with `www-data` (or with the same uid)

**Comment by bshaffer** [comment]
@tmatsuo okay, sounds good. This script should still be run as root though!

**Pull request merged**